lib/cli.ex

```diff
@@ -1,7 +1,28 @@
 defmodule Onigumo.CLI do
-  def main([component]) do
-    module = Module.safe_concat("Onigumo", component)
-    root_path = File.cwd!()
-    module.main(root_path)
+  def main(argv) do
+    {parsed_switches, _, _} =
+      argv
+      |> parse_argv
+
+    module =
+      parsed_switches
+      |> Keyword.get(:component, "Downloader")
+      |> safe_concat()
```

Comment:
We save the result to variable with name `module`, but the switch to be parse is called `component`. Would not be fancier when switch would be called as `module` also?

Reply:
The variable name `module` is really module -> `Onigumo.Donwloader` is module and it is the thing we get from pipes to `module` variable.